contrib/interconnect/ic_modules.c

```diff
@@ -146,20 +146,15 @@ _PG_init(void)
 				 errmsg("could not load interconnect outside process shared preload")));
     }
 
-    switch(Gp_interconnect_type) {
-        case INTERCONNECT_TYPE_TCP:
-            CurrentMotionIPCLayer = &tcp_ipc_layer;
-            break;
-        case INTERCONNECT_TYPE_UDPIFC:
-            CurrentMotionIPCLayer = &udpifc_ipc_layer;
-            break;
-        case INTERCONNECT_TYPE_PROXY:
-            CurrentMotionIPCLayer = &proxy_ipc_layer;
-            break;    
-        default:
-            ereport(ERROR,
-				(errcode_for_file_access(),
-				 errmsg("could not decide interconnect type")));
-    }
-    
+	IPCLayerImpls[CurrentIPCLayerImplNum++] = &tcp_ipc_layer;
+	IPCLayerImpls[CurrentIPCLayerImplNum++] = &udpifc_ipc_layer;
+	IPCLayerImpls[CurrentIPCLayerImplNum++] = &proxy_ipc_layer;
+
+	/*
+	 * set CurrentMotionIPCLayer with Gp_interconnect_type if it's NULL.
+	 */
+	if (CurrentMotionIPCLayer == NULL)
+	{
+		SetCurrentMotionIPCLayer(Gp_interconnect_type);
+	}
```

Comment:
If the ic_type is set to other value(implement ic in other module), setting `CurrentMotionIPCLayer` will fail, so the database can't startup.
Better to:
1. Set `CurrentMotionIPCLayer` later, after all modules are initialized, so all possible ic implementations are added.
2. If the desired ic_type is not found, choose one of the registered ic impl, log a warning message, instead of failing to startup.

Reply:
fix
1. Set CurrentMotionIPCLayer after interconnect.so is loaded;
2. ereport(WARNING, ...) if ic_type is not found;